valohai_yaml/objs/task.py

```diff
@@ -0,0 +1,94 @@
+from enum import Enum
+from typing import Any, List, Optional, Union
+
+from valohai_yaml.objs.base import Item
+from valohai_yaml.objs.utils import (
+    check_type_and_listify,
+    consume_array_of,
+)
+
+
+class TaskParameterType(Enum):
+    """Represents a task parameter style definition."""
+
+    BAYESIAN_TPE = 'bayesian_tpe'
+    GREED_SEARCH = 'greed_search'
+    DISTRIBUTED = 'distributed'
+    RANDOM_SEARCH = 'random_search'
+    MANUAL_SEARCH = 'manual_search'
+
+    @classmethod
+    def cast(cls, value: Optional[Union['TaskParameterType', str]]) -> 'TaskParameterType':
+        if not value:
+            return TaskParameterType.GREED_SEARCH
+        if isinstance(value, TaskParameterType):
+            return value
+        value = str(value).lower()
+        return TaskParameterType(value)
+
+
+class VariantParameterStyle(Enum):
+    """Represents a variant parameter style definition."""
+
+    LOGSPACE = 'logspace'
+    MULTIPLE = 'multiple'
+    LINEAR = 'linear'
+    SINGLE = 'single'
+    RANDOM = 'random'
+
+    @classmethod
+    def cast(cls, value: Optional[Union['VariantParameterStyle', str]]) -> 'VariantParameterStyle':
+        if not value:
+            return VariantParameterStyle.SINGLE
+        if isinstance(value, VariantParameterStyle):
+            return value
+        value = str(value).lower()
+        return VariantParameterStyle(value)
+
+
+class VariantParameter(Item):
```

Comment:
What do you think, at some point we want to move these from Roi to Valohai-yaml fully. As for example we now use Parameters for Executions from valohai-yaml library. Should that be a part of this PR or should we do it later?

Reply:
frankly I don't think it is a good idea to do more then one things in one pull request. I would do it in another PR

Reply:
Sounds good, just make sure to read through the current implementation in Roi to make sure there isn't any blockers in this version.